include/ccmath/internal/builtins/fma.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <type_traits>
+
+namespace ccm::internal::builtin
+{
+	template <typename T>
+	inline constexpr bool has_constexpr_fma =
+		#if defined(__GNUC__) && (__GNUC__ > 6 || (__GNUC__ == 6 && __GNUC_MINOR__ >= 1)) && !defined(__clang__)
+			(std::is_same_v<T, float> || std::is_same_v<T, double> || std::is_same_v<T, long double>);
+	#else
+				false;
+	#endif
+
+	template <typename T>
+	constexpr auto fma(T x, T y, T z)
```

Comment:
For now you're checking three types: float, double and long double. Maybe add a static assert just to be sure?